datanode/gateway/graphql/tools/tools.go

```diff
@@ -10,8 +10,12 @@
 // of this software will be governed by version 3 or later of the GNU General
 // Public License.
 
+//go:build tools
+// +build tools
+
 package tools
 
 import (
-	_ "github.com/99designs/gqlgen/graphql" // this presumably does something useful
+	_ "github.com/99designs/gqlgen"
+	_ "github.com/99designs/gqlgen/graphql/introspection"
```

Comment:
Yes, these are the fixes I mentioned for gqlgen. The gqlgen dependency was updated recently and without them, when you run gqlgen you get errors complaining about the urfave/cli library.

Reply:
The import for `github.com/urfave/cli/v2' in `tools.go` wasn't in the branch I had and the changes I made was what was recommended in the gqlgen readme.

For now I've reversed out those imports and merged in your changes to `tools.go` so that there's less confusion in the git history.

Reply:
well if you say it's working go on. I initially imported gqlgen like so at first, and it broke the lints as the root is the main package, which you're not allowed to import.